src/melee.cpp

```diff
@@ -151,7 +151,7 @@ int player::get_hit_weapon( const item &weap ) const
     ///\EFFECT_BASHING improves hit chance for bashing weapons
     ///\EFFECT_CUTTING improves hit chance for cutting weapons
     ///\EFFECT_STABBING improves hit chance for piercing weapons
-    auto bonus = get_skill_level( weap.melee_skill() );
+    auto bonus = get_skill_level( is_armed() ? weap.melee_skill() : skill_unarmed );
```

Comment:
Would be simpler and more robust to just allow `is_null` items to count as unarmed instead of special casing it everywhere.